sui_storage/src/indexes.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! IndexStore supports creation of various ancillary indexes of state in SuiDataStore.
+//! The main user of this data is the explorer.
+
+use rocksdb::Options;
+use serde::{de::DeserializeOwned, Serialize};
+use std::collections::HashMap;
+use std::path::Path;
+use sui_types::base_types::{ObjectID, ObjectRef, SuiAddress, TransactionDigest};
+use sui_types::batch::TxSequenceNumber;
+
+use sui_types::error::SuiResult;
+use sui_types::messages::InputObjectKind;
+use sui_types::object::Object;
+
+use typed_store::rocks::DBMap;
+use typed_store::{reopen, traits::Map};
+
+pub struct IndexStore {
+    /// Index from sui address to transactions initiated by that address.
+    transactions_from_addr: DBMap<(SuiAddress, TxSequenceNumber), TransactionDigest>,
+
+    /// Index from sui address to transactions that were sent to that address.
+    transactions_to_addr: DBMap<(SuiAddress, TxSequenceNumber), TransactionDigest>,
+
+    /// Index from object id to transactions that used that object id as input.
+    transactions_by_input_object_id: DBMap<(ObjectID, TxSequenceNumber), TransactionDigest>,
+
+    /// Index from object id to transactions that modified/created that object id.
+    transactions_by_mutated_object_id: DBMap<(ObjectID, TxSequenceNumber), TransactionDigest>,
+}
+
+impl IndexStore {
+    pub fn open<P: AsRef<Path>>(path: P, db_options: Option<Options>) -> Self {
+        let mut options = db_options.unwrap_or_default();
+
+        /* The table cache is locked for updates and this determines the number
+           of shareds, ie 2^10. Increase in case of lock contentions.
+        */
```

Comment:
Don't use multi-line comments, prefer single line comments `//` which is the standard rust style

Reply:
Sorry, copy-pasted this from authority_store.rs